### DISCUSSION TITLE
Experience & Applications

### DISCUSSION THREAD

**Discussion by jakegros** [USU-ITLS-x265-23-summer/class-repo#36] [Week 3 - Discussions]
What kind of experience do you have in this area? 
Were you just curious about this course or do you have specific applications you're hoping to gain?
What processes have you tried to solve rubber-ducky style?

**Comment by coopsand** [comment]
Hi Jacob!

I have had some experience programming a little bit before. I have programmed in Java, Python, and C++. (And now I guess I have done some HTML and JS.) 

I want to work as a UX/UI Designer. Knowing some basic HTML and JavaScript will allow me to better interact with the actual developers of the designs. Knowing some HTML will also allow me to build more significant prototypes of my designs. I hope to continue to learn how to build sites using HTML and especially style it using CSS elements. That is probably the most important and significant thing for me to know how to do.

When it comes down to debugging with a rubber-ducky, I actually have a physical rubber duck that I got from a CS class that I have taken. A lot of times, it helps to just talk through the problems that you are facing. I've found that sometimes, something clicks in my brain and I know what to do. Another great resource is if you have any friends or connections that can help walk you through a process. Posting questions in our discussions will allow us to work together to problem solve and collaborate, so take advantage of that as well.

I hope that gave you some insight. I'd love to hear your answers to the questions you asked. :)

**Comment by kassyoveson** [comment]
Back in May, I had no experience with coding. This course is required for my major, which is Human Experience Design Interaction. I know that by having knowledge in JS, CSS and HTML will help me as an UX/UI designer. At the end of July, I feel like I got my feet wet, but I still don't speak fluent JS, CSS or HTML. One of my experiences with a 'rubber-duck style' was when we were trying to learn CSS for the first time. I didn't know what they wanted us to type in the 'plates' exercise. I was in a hotel lobby, so talking to myself probably seemed really weird. I was able to work through it, but it took me a little while.

**Comment by gabebott** [comment]
I have a little experience in prior to this class with coding. I was previously a finance major and took multiple classes on data analytics. In these classes I was able to learn the basics of SQL and other programs. When I switched majors I took some other beginner coding classes, but this is my first time really diving into HTML which I am very excited to continue learning about as I think it will assist me in my future career in UX Design.
